pkgs/os-specific/bsd/freebsd/patches/14.0/localedef.patch

```diff
@@ -0,0 +1,209 @@
+diff --git a/include/_ctype.h b/include/_ctype.h
+index 91e6b1d14f6b..a6896b598da3 100644
+--- a/include/_ctype.h
++++ b/include/_ctype.h
+@@ -44,7 +44,7 @@
+ #define	__CTYPE_H_
+ 
+ #include <sys/cdefs.h>
+-#include <sys/_types.h>
++#include <sys/types.h>
+ 
+ #define	_CTYPE_A	0x00000100L		/* Alpha */
+ #define	_CTYPE_C	0x00000200L		/* Control */
+diff --git a/lib/libc/locale/collate.h b/lib/libc/locale/collate.h
+index 2d3723b49f5b..6bbff732b9d7 100644
+--- a/lib/libc/locale/collate.h
++++ b/lib/libc/locale/collate.h
+@@ -36,6 +36,7 @@
+ #ifndef _COLLATE_H_
+ #define	_COLLATE_H_
+ 
++#include <stdint.h>
+ #include <sys/cdefs.h>
+ #include <sys/types.h>
+ #include <limits.h>
+diff --git a/usr.bin/localedef/charmap.c b/usr.bin/localedef/charmap.c
+index 44b7e3292eae..79c30b7cf372 100644
+--- a/usr.bin/localedef/charmap.c
++++ b/usr.bin/localedef/charmap.c
+@@ -31,6 +31,7 @@
+ /*
+  * CHARMAP file handling for localedef.
+  */
++#include <stdint.h>
+ #include <sys/cdefs.h>
+ #include <sys/types.h>
+ #include <sys/tree.h>
+@@ -70,13 +71,49 @@ static const struct {
+ 	int	ch;
+ } portable_chars[] = {
+ 	{ "NUL",		'\0' },
++	{ "SOH",		'\x01' },
++	{ "STX",		'\x02' },
++	{ "ETX",		'\x03' },
++	{ "EOT",		'\x04' },
++	{ "ENQ",		'\x05' },
++	{ "ACK",		'\x06' },
++	{ "BEL",		'\a' },
+ 	{ "alert",		'\a' },
++	{ "BS",			'\b' },
+ 	{ "backspace",		'\b' },
++	{ "HT",			'\t' },
+ 	{ "tab",		'\t' },
+-	{ "carriage-return",	'\r' },
++	{ "LF",			'\n' },
+ 	{ "newline",		'\n' },
++	{ "VT",			'\v' },
+ 	{ "vertical-tab",	'\v' },
++	{ "FF",			'\f' },
+ 	{ "form-feed",		'\f' },
++	{ "CR",			'\r' },
++	{ "carriage-return",	'\r' },
++	{ "SO",			'\x0e' },
++	{ "SI",			'\x0f' },
++	{ "DLE",		'\x10' },
++	{ "DC1",		'\x11' },
++	{ "DC2",		'\x12' },
++	{ "DC3",		'\x13' },
++	{ "DC4",		'\x14' },
++	{ "NAK",		'\x15' },
++	{ "SYN",		'\x16' },
++	{ "ETB",		'\x17' },
++	{ "CAN",		'\x18' },
++	{ "EM",			'\x19' },
++	{ "SUB",		'\x1a' },
++	{ "ESC",		'\x1b' },
++	{ "FS",			'\x1c' },
++	{ "IS4",		'\x1c' },
++	{ "GS",			'\x1d' },
++	{ "IS3",		'\x1d' },
++	{ "RS",			'\x1e' },
++	{ "IS2",		'\x1e' },
++	{ "US",			'\x1f' },
++	{ "IS1",		'\x1f' },
++	{ "DEL",		'\x7f' },
```

Comment:
What's going on with all these additions?

Reply:
@artemist this is your patch

Reply:
Based on my investigation, this doesn't actually impact the building of locales as we currently have it configured, and only adjusts the default charmap, which is used if you don't specify any charmap file at generation time. Diff indicates that there's no actual difference between the locales derivation with or without this patch, so I'm removing it.